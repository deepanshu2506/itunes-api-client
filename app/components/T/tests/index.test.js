/**
 *
 * Tests for T
 *
 */

import React from 'react';
import { renderWithIntl, getComponentStyles } from '@utils/testUtils';
import { T } from '../index';

describe('<T /> component tests', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<T />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 T component', () => {
    const { getAllByTestId } = renderWithIntl(<T />);
    expect(getAllByTestId('t').length).toBe(1);
  });

  it('should contain render the text according to the id', () => {
    const { getAllByText } = renderWithIntl(<T id="iTunes_search" />);
    expect(getAllByText(/iTunes Search/).length).toBe(1);
  });

  it('should have a margin-bottom of 5px when we pass marginBottom as 5', () => {
    const props = {
      marginBottom: 5,
      id: 'iTunes_search'
    };
    const styles = getComponentStyles(T, props);
    expect(styles['margin-bottom']).toBe(`${props.marginBottom}px`);
  });
  it('should render a span if the span prop is true', () => {
    const { getAllByText } = renderWithIntl(<T id="iTunes_search" span />);
    expect(getAllByText(/iTunes Search/)[0].tagName).toBe('SPAN');
  });
});
