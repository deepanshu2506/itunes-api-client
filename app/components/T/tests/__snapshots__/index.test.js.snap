// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<T /> component tests should render and match the snapshot 1`] = `
<body>
  <div>
    <p
      class="T__StyledParagraph-gjlic1-0 bhyTwG"
      data-testid="t"
    />
  </div>
</body>
`;
